pgrx/src/memcx.rs

```diff
@@ -16,27 +18,35 @@ pub struct MemCx<'mcx> {
 }
 
 impl<'mcx> MemCx<'mcx> {
-    /// You probably shouldn't be using this.
+    /// Wrap the provided [`pg_sys::MemoryContext`]
+    ///
+    /// # Safety:
+    /// Assumes the provided [`pg_sys::MemoryContext`] is valid and properly initialized.
+    /// This method does check to ensure the pointer is non-null, but that is the only sanity
+    /// check that is performed.
     pub(crate) unsafe fn from_ptr(raw: pg_sys::MemoryContext) -> MemCx<'mcx> {
         let ptr = NonNull::new(raw).expect("memory context must be non-null");
         MemCx { ptr, _marker: PhantomData }
     }
 
-    /// You probably shouldn't be using this.
-    pub(crate) unsafe fn alloc_bytes(&self, size: usize) -> *mut u8 {
+    /// Allocate a raw byte buffer `size` bytes in length
+    /// and returns a pointer to the new allocation.
+    pub unsafe fn alloc_bytes(&self, size: usize) -> *mut u8 {
         pg_sys::MemoryContextAlloc(self.ptr.as_ptr(), size).cast()
     }
 
-    /// You probably shouldn't be using this.
-    pub(crate) unsafe fn exec_in<T>(&self, f: impl FnOnce() -> T) -> T {
+    /// Stores the current global memory context, switches to *this* memory context,
+    /// and executes the closure `f`.
+    /// Once `f` completes, the previous global memory context is restored.
+    pub unsafe fn exec_in<T>(&self, f: impl FnOnce() -> T) -> T {
         let remembered = pg_sys::MemoryContextSwitchTo(self.ptr.as_ptr());
         let res = f();
         pg_sys::MemoryContextSwitchTo(remembered);
         res
     }
 }
 
-/// Acquire the current context and operate inside it.
+/// Acquire the current global context and operate inside it.
```

Comment:
Not sure what this means? Memory contexts are usually not conceptually global, but rather bound to increasingly local scopes. The current context is often one which will end immediately after the outer function call is over.

Reply:
Should be corrected in https://github.com/pgcentralfoundation/pgrx/pull/1737/commits/f149072c318f7e39c212daf3b59e0f723d1dfd46